src/main/java/io/github/oliviercailloux/y2018/apartments/valuefunction/LinearAVF.java

```diff
@@ -0,0 +1,1083 @@
+            && nbBathroomsValueFunction.getSubjectiveValue((double) apart.getNbBathrooms()) <= 1,
+        "The subjective value of the number of bathrooms should be between 0 and 1");
+    nbBathroomsSubjectiveValue =
+        nbBathroomsValueFunction.getSubjectiveValue((double) apart.getNbBathrooms());
+    LOGGER.debug(
+        "The number of bathrooms subjective value has been set to {}", nbBathroomsSubjectiveValue);
+
+    checkArgument(
+        terraceValueFunction.getSubjectiveValue(apart.getTerrace()) >= 0
+            && terraceValueFunction.getSubjectiveValue(apart.getTerrace()) <= 1,
+        "The subjective value of the terrace should be between 0 and 1");
+    terraceSubjectiveValue = terraceValueFunction.getSubjectiveValue(apart.getTerrace());
+    LOGGER.debug("The terrace subjective value has been set to {}", terraceSubjectiveValue);
+
+    checkArgument(
+        floorAreaTerraceValueFunction.getSubjectiveValue(apart.getFloorAreaTerrace()) >= 0
+            && floorAreaTerraceValueFunction.getSubjectiveValue(apart.getFloorAreaTerrace()) <= 1,
+        "The subjective value of the floor area of the terrace should be between 0 and 1");
+    floorAreaTerraceSubjectiveValue =
+        floorAreaTerraceValueFunction.getSubjectiveValue(apart.getFloorAreaTerrace());
+    LOGGER.debug(
+        "The floor area of the terrace subjective value has been set to {}",
+        floorAreaTerraceSubjectiveValue);
+
+    checkArgument(
+        wifiValueFunction.getSubjectiveValue(apart.getWifi()) >= 0
+            && wifiValueFunction.getSubjectiveValue(apart.getWifi()) <= 1,
+        "The subjective value of the wifi should be between 0 and 1");
+    wifiSubjectiveValue = wifiValueFunction.getSubjectiveValue(apart.getWifi());
+    LOGGER.debug("The wifi subjective value has been set to {}", wifiSubjectiveValue);
+
+    checkArgument(
+        pricePerNightValueFunction.getSubjectiveValue(apart.getPricePerNight()) >= 0
+            && pricePerNightValueFunction.getSubjectiveValue(apart.getPricePerNight()) <= 1,
+        "The subjective value of the price per night should be between 0 and 1");
+    pricePerNightSubjectiveValue =
+        pricePerNightValueFunction.getSubjectiveValue(apart.getPricePerNight());
+    LOGGER.debug(
+        "the price per night subjective value has been set to {}", pricePerNightSubjectiveValue);
+
+    checkArgument(
+        nbMinNightValueFunction.getSubjectiveValue((double) apart.getNbMinNight()) >= 0
+            && nbMinNightValueFunction.getSubjectiveValue((double) apart.getNbMinNight()) <= 1,
+        "The subjective value of the minimum number of nights should be between 0 and 1");
+    nbMinNightSubjectiveValue =
+        nbMinNightValueFunction.getSubjectiveValue((double) apart.getNbMinNight());
+    LOGGER.debug(
+        "The minimum number of nights subjective value has been set to {}",
+        nbMinNightSubjectiveValue);
+
+    checkArgument(
```

Comment:
Je ne conçois pas ceci comme un problème d’argument : c’est la fonction qui doit garantir les bornes, par l’appartement. C’est donc plutôt un problème de State en cas de violation.
Il faut généraliser cette logique, car ce code fait littéralement dix fois la même chose.

Reply:
fixed in  adec0154ecafb1a1f219982de3221f85e8ffe8d0